cranelift/2024/cranelift-07-31.md

```diff
@@ -19,5 +19,47 @@
 
 ### Attendees
 
+- fitzgen
+- alexcrichton
+- uweigand
+- elliottt
+- avanhatt
+- abrown
+- cfallin
+
 ### Notes
 
+- Pointer provenance and Pulley
+  - alexcrichton: recap: Pulley has arbitrary loads/stores from integer
+    addresses; doesn't work with strict provenance
+  - turns out Rust won't require strict provenance; breaks lots of things
+  - under loose provenance, we do need to mark pointers that escape to Pulley
+  - fitzgen: things we need to do for escapes to Pulley are the same we do for JIT code
+  - cfallin: PCC provides the same thing with memory types -- could use this as
+    a way to get provenance in the future
+  - abrown: how did we get on this quest?
+    - alexcrichton: originally, UB bug found by miri; then Pulley, getting
+      tests running in miri
+
+- liveness analysis and user stack maps
+  - fitzgen: all tests passing with user stack maps now, but debugging a
+    fuzzing failure in table-ops test
+  - fundamentally need either a fixup pass or full fixpoint loop for liveness
+  - cfallin: third way: IonMonkey regalloc's approximation: when we see a loop
+    edge, take the loop end's liveness and smear it over the whole loop body
+
+- updates
+  - jameysharp: Cranelift libcalls and stack limit checks; setting on Cranelift
+    to tell it to emit stack-limit checks in such cases
+  - elliottt: no updates
+  - avanhatt: no updates
+  - abrown: no updates
+  - alexcrichton: no updates
+  - uweigand: done with tail calls on s390x! PR pending
+    - [something about spilltmp register and register choices; missed most due
+      to connection issues]
```

Comment:
my connection dropped out here for a bit; @uweigand @jameysharp @elliottt @fitzgen if you want to fill in some more here please feel free!

Reply:
Indirect tail calls need a register to hold the target.  But this has to be distinct from all callee-saved registers (which will be restored before the tail call) and all argument registers used by the call, which might mean no allocatable register is available  at all for this purpose.   In this case I'm using the spilltmp, which requires an extra move.  But there's probably no good way around that.

Reply:
Thanks!